### PR TITLE
[#211] Refactor customizer to use files

### LINKED ISSUE #211: Colorizer should not target ::root, not body
We don't want to change the specificity when we set the color variables.  We just need to ensure the colorizer code is further down in the source order.

### PR DESCRIPTION
**Pull request by shadcn**
Fixes #211 

This PR refactors the customizer to save to a generated CSS file instead of inline style. The generated CSS file is then put into a library that the module creates on behalf of the customizable theme.

This allows the weight of the file to be set and can be moved down to overwrite any variables set by other files (example the theme CSS file).

As a side effect, caching of the file is handled for us 👍 .

I added an event subscriber to re-generate the CSS file when config is updated. This way if you change values on local and import to another environment, an updated CSS file is created for you.

### CODE REVIEW

File: modules/custom/apigee_kickstart_customizer/src/Customizer.php [hunk 1]
```diff
@@ -113,6 +147,31 @@ public function getDefinitionForTheme($theme = NULL): array {
...
+   * {@inheritdoc}
+   */
+  public function updateStylesheetForTheme($theme = NULL): void {
+    $theme = $theme ?? $this->activeTheme->getName();
+    $values = $this->configFactory->get('apigee_kickstart_customizer.theme.' . $theme)
+      ->get('values');
+
+    // TODO: Add scope support.
+    $css = ':root {';
+    foreach ($values as $name => $value) {
```
Comment (jacine) [inline modules/custom/apigee_kickstart_customizer/src/Customizer.php:160]: I'm getting an error here when I try to delete this config.  Should be able to delete this, and stop the file write, etc.

Comment (shadcn) [reply modules/custom/apigee_kickstart_customizer/src/Customizer.php:160]: Pushed a fix for it. On config delete, the stylesheet is deleted.

Comment (jacine) [reply modules/custom/apigee_kickstart_customizer/src/Customizer.php:160]: Yep, works perfectly!  Thanks @arshad :)

**Review by jacine — APPROVED** [review]
This is awesome @arshad! 🎨 👏 

I think we _might_ need an update hook, but we can do that in a follow up if it's needed. Going to merge this.

**Pull request merged**